server/datastore/mysql/mdm.go

```diff
@@ -26,3 +28,62 @@ END AS platform
 
 	return p, nil
 }
+
+func (ds *Datastore) ListMDMCommands(
+	ctx context.Context,
+	tmFilter fleet.TeamFilter,
+	listOpts *fleet.MDMCommandListOptions,
+) ([]*fleet.MDMCommand, error) {
+	appleStmt := `
+SELECT
+    nvq.id as host_uuid,
+    nvq.command_uuid,
+    COALESCE(NULLIF(nvq.status, ''), 'Pending') as status,
+    COALESCE(nvq.result_updated_at, nvq.created_at) as updated_at,
+    nvq.request_type,
+    h.hostname,
+    h.team_id
+FROM
+    nano_view_queue nvq
+INNER JOIN
+    hosts h
+ON
+    nvq.id = h.uuid
+WHERE
+   nvq.active = 1
+`
+
+	// TODO(roberto): note how we're not filtering by `active = 1` on windows.
+	// Nano behaves like this:
+	// - When a device is unenrolled, all pending commands are set `active = 0`
+	// - When a command is delivered, the entry is removed from the table
+	//
+	// In Windows, we set `active = 0` when we get the results of the command. What we should do?
```

Comment:
I think if we are going to use `active` in both tables we should make `active` mean the same thing for both Windows and macOS. We don't have any real need currently to keep track of `active` for Windows in the same sense as nano does. So I think it would probably be better at this point to rename the `active` column in the Windows table to something other than `active` so it is clear that it has a different meaning.

Reply:
agreed, I think maybe either delete or soft delete once we receive the command will do, and we make `active` work like it does for nano

Reply:
@mna @gillespi314 I started implementing what's described above.

The note above is wrong. nano sets `active = 0` when a device re-enrolls ([source](https://github.com/micromdm/nanomdm/blob/3abd0dc82a3cb40e2fc4da42a2d9659570da9507/service/nanomdm/service.go#L118-L123))

```
	// clear the command queue for any enrollment or sub-enrollment.
	// this prevents queued commands still being queued after device
	// unenrollment.
```

However, I don't think we need to do this? when a device enrolls we delete the device:

https://github.com/fleetdm/fleet/blob/f677b20b6c935969a807e98718f4cf8c767d35f3/server/service/microsoft_mdm.go#L1106-L1110

Which causes a delete on cascade in the command tables.

Whether or not we want to delete the device on re-enrollment is another can of worms.

Thinking about removing `active` altogether, any thoughts?

Reply:
I have pushed a draft PR if we decide to remove this column: https://github.com/fleetdm/fleet/pull/14826

Reply:
@roperzh 

> Thinking about removing active altogether, any thoughts?

Yeah I think it was a "nice to have" that we let through because it was in nano/apple too, but we don't strictly need it - only thing is we'll have no mechanism to disable a command queued for a Windows host, but then again it would've been a manual SQL to run, might as well delete it if this need happens.

Thinking out loud, it shouldn't cause any issues with re-sending commands, because the "pending" commands are those that have not received any response yet, so it should be fine without the `active` column.